src/Aspire.Hosting.Orleans/IDistributedApplicationBuilderExtensions.cs

```diff
@@ -0,0 +1,146 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Orleans.Shared;
+
+namespace Aspire.Hosting;
+
+/// <summary>
+/// Extensions to <see cref="IDistributedApplicationBuilder"/> related to Orleans.
+/// </summary>
+public static class IDistributedApplicationBuilderExtensions
+{
+    private const string OrleansConfigKeyPrefix = "Orleans";
+
+    public static IResourceBuilder<OrleansResource> AddOrleans(
+        this IDistributedApplicationBuilder builder,
+        string name)
+        => builder.AddResource(new OrleansResource(name));
+
+    public static IResourceBuilder<OrleansResource> WithClusterId(
```

Comment:
If ClusterId is required, then we should take it as an argument on `AddOrleans(...)`.

Reply:
`ClusterId` could be set by a membership provider (for example the `Localhost` one)

Reply:
We generate one for you if you don't specify one. This is to prevent runs from clashing with each other.
If we could generate a random one at dev time and select a fixed one at prod time (eg, the app name, or don't set one at all so it uses "default", I think that would be better.